### DISCUSSION TITLE
How does type safety work with this package?

### DISCUSSION THREAD

**Discussion by vim-diesel** [next-safe-action/next-safe-action#193] [Q&A] [answered]
I am just wondering how it works, because after I have everything setup in my actions file, it seems I can enter whatever I want in the client component when calling the action:

```
  return (
    <form
      onSubmit={async (e: FormEvent) => {
        e.preventDefault();
        const formData = new FormData(e.target as HTMLFormElement);

        const res = await executeAsync({
          title: formData.get('title'),
          startDate: startDate?.toISOString(),
          endDate: endDate?.toISOString(),
          location: formData.get('location'),
          price: Number(formData.get('price')),
          status: formData.get('eventStatus'),
        });

        console.log('result', res);
      }}
    >
      <div className='space-y-12'>
...
```

I get no warnings when putting a number in the title, for example, when that is not what the schema defines:

```
const executeAsync: (input: any) => Promise<SafeActionResult<string, ZodObject<{
    title: ZodString;
    startDate: ZodString;
    endDate: ZodOptional<ZodString>;
    location: ZodOptional<ZodString>;
    price: ZodOptional<...>;
    status: ZodOptional<...>;
```

Am I wrong in assuming that the type safety would extend to calling the function and not just in the action file, where it seems to work?

**Comment by vim-diesel** [comment]
I should add, I only mean when calling the function and defining its inputs, I don't mean when I run the code, all the validation still works.

**Comment by TheEdoRan** [comment, marked as answer]
Well it sure isn't the intended behavior:

![type mismatch error](https://github.com/TheEdoRan/next-safe-action/assets/1337629/f62eef39-68ad-4f71-834f-2f7064ef8f84)

Are you perhaps using TypeScript 5.5? If this is the case, please check out this issue and comment:

- https://github.com/TheEdoRan/next-safe-action/issues/180#issuecomment-2201607407

**Reply by vim-diesel** [reply]
Brilliant! Thanks for your comment.

**Reply by TheEdoRan** [reply]
Great!
